pdns/dnsdistdist/dnsdist-tcp-downstream.cc

```diff
@@ -760,22 +800,34 @@ bool TCPConnectionToBackend::isXFRFinished(const TCPResponse& response, TCPQuery
         }
         auto raw = unknownContent->getRawContent();
         auto serial = getSerialFromRawSOAContent(raw);
-        ++query.d_xfrSerialCount;
         if (query.d_xfrMasterSerial == 0) {
           // store the first SOA in our client's connection metadata
-          ++query.d_xfrMasterSerialCount;
           query.d_xfrMasterSerial = serial;
+          if (query.d_xfrMasterSerial == query.d_xfrQuerySerial) {
```

Comment:
Should this be `query.d_xfrMasterSerial <= query.d_xfrQuerySerial` as per the RFC citation below? (If not, maybe comment why?)

Reply:
Yes, it should, many thanks for spotting that!